File: swift3_For-Loop/DSL/DSL/ViewController.m
//
//  ViewController.m
//  DSL
//
//  Created by licong on 2017/2/27.
//  Copyright © 2017年 Richard. All rights reserved.
//

//http://chuansong.me/n/1554838228427

#import "ViewController.h"
#import "ViewMaker.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    //关于数据常量区的测试
    NSString *s0 = @"hello";
    NSString *s1 = @"hello";
    NSString *s2 = [NSString stringWithFormat:@"hello"];
    NSString *s3 = [NSString stringWithFormat:@"hello"];
    NSString *s4 = [NSString stringWithFormat:@"hello"];
    NSString *s5 = [NSString stringWithString:s0];
    NSString *s6 = [NSString stringWithFormat:@"%@", s1];
    

    NSLog(@"s0->%p",s0);
    NSLog(@"s1->%p",s1);
    NSLog(@"s2->%p",s2);
    NSLog(@"s3->%p",s3);
    NSLog(@"s4->%p",s4);
    NSLog(@"s5->%p",s5);
    NSLog(@"s6->%p",s6);


    NSLog(@"s0 vaule->%@",s0);
    NSLog(@"s1 vaule->%@",s1);
    NSLog(@"s2 vaule->%@",s2);
    NSLog(@"s3 vaule->%@",s3);
    NSLog(@"s4 vaule->%@",s4);
    NSLog(@"s5 vaule->%@",s5);
    NSLog(@"s6 vaule->%@",s6);
    
    
    NSLog(@"s0 <-> s2 %d", s0 == s2);
    NSLog(@"s2 <-> s3 %d", s2 == s3);

    
    NSLog(@"s0 Equal s2 %d", [s0 isEqualToString: s2]);
    NSLog(@"s2 Equal s3 %d", [s2 isEqualToString: s3]);

    //总结isEqualToString是比较值,==号是比较指针。
    //stringWithFormat生成的不再数据常量去
    //stringWithString生成得在数据常量区
    
    
    
    [ViewMaker makeView].with.position(80,80).size(100,100).bgColor([UIColor blueColor]).intoView(self.view);
}



- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: App支持搜索/APPSearch/APPSearch/ViewController.swift
//
//  ViewController.swift
//  APPSearch
//
//  Created by licong on 2017/3/21.
//  Copyright © 2017年 Richard. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    var table = UITableView()
    var dataArray = [Person]()
    override func viewDidLoad() {
        super.viewDidLoad()
        table.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: "identifier")
        view.addSubview(table)
        table.frame = UIScreen.main.bounds
        table.delegate = self
        table.dataSource = self
        let dataSource = DataSource()
        dataArray = dataSource.dataList
        
        dataSource.savePeople()
        table.reloadData()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}


extension ViewController: UITableViewDelegate, UITableViewDataSource {
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataArray.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "identifier", for: indexPath)
        let object =  dataArray[indexPath.row]
        cell.textLabel?.text = object.name
        cell.imageView?.image = UIImage(named: object.icon)
        return cell
    }
}
